subprojects/base-services/src/main/java/org/gradle/internal/hash/FileHasher.java

```diff
@@ -24,7 +24,7 @@
 
 public interface FileHasher {
     /**
-     * Returns the hash of the given input stream.
+     * Returns the hash of the given input stream. This method buffers reads on the stream.
```

Comment:
I don't think Javadoc is the right place for this at all. I'd rather solve this via the type system, like expect a `BufferedStream` when external parties are supposed to handle the buffering or something (which should be pretty rare). Buffering and closing streams is one big pile of complexity within Java that we regularly end up misusing.

Until then we can leave this in place, but I think we're better off without it.